sdk/communication/azure-communication-email/azure/communication/email/_email_client.py

```diff
@@ -0,0 +1,103 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from uuid import uuid4
+from azure.core.tracing.decorator import distributed_trace
+from ._shared.utils import parse_connection_str, get_current_utc_time
+from ._shared.policy import HMACCredentialsPolicy
+from ._generated._azure_communication_email_service import AzureCommunicationEmailService
+from ._version import SDK_MONIKER
+from ._generated.models import SendEmailResult, SendStatusResult, EmailMessage
+
+class EmailClient(object): # pylint: disable=client-accepts-api-version-keyword
+    """A client to interact with the AzureCommunicationService Email gateway.
+
+    This client provides operations to send an email and monitor its status.
+
+    :param str endpoint:
+        The endpoint url for Azure Communication Service resource.
+    :param TokenCredential credential:
+        The TokenCredential we use to authenticate against the service.
+    """
+    def __init__(
+            self,
+            endpoint, # type: str
+            credential, # type: TokenCredential
+            **kwargs # type: Any
+        ):
+        # type: (...) -> None
+        authentication_policy = HMACCredentialsPolicy(endpoint, credential)
+
+        self._generated_client = AzureCommunicationEmailService(
+            endpoint,
+            authentication_policy=authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            **kwargs
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str, # type: str
+        **kwargs # type: Any
+    ): # type: (...) -> EmailClient
+        """Create EmailClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Communication Service resource.
+        :returns: Instance of EmailClient.
+        :rtype: ~azure.communication.EmailClient
+        """
+        endpoint, access_key = parse_connection_str(conn_str)
+
+        return cls(endpoint, access_key, **kwargs)
```

Comment:
This shouldn't pass in the string directly - it should be wrapped in a `azure.core.credential.AzureKeyCredential`
It probably means that the constructor `credential` parameter should be type hinted as
`Union[TokenCredential, AzureKeyCredential]`